### DISCUSSION TITLE
使用FormGrid，如何让组件独占一行

### DISCUSSION THREAD

**Discussion by beita-wang** [alibaba/formily#3858] [Q&A 2.x]
![4_F(HE6H05X9Y)E@M{X6Q{M](https://github.com/alibaba/formily/assets/58249740/840e2335-856e-4d46-8715-d07a721f34cd)

如上图所示，我设置FormGrid  maxColumns为4  假设图中两个元素，我如果想让richtextBox独占一行  给他gridSpan设为4，textarea设为2，但是richtextBox没挤下去。如果不给textarea设为4，怎么通过设置richtextBox来让richtextBox独占一行，而不是设置前一个元素把后一个元素挤下去。

**Comment by Landon-CN** [comment]
grid内部会自动计算并填充空隙，可以给grid传入 strictAutoFit规避
```ts
 const span = grid.options.strictAutoFit
      ? targetSpan
      : targetSpan > remainColumns
      ? remainColumns
      : targetSpan
```

**Reply by beita-wang** [reply]
我这边已经传入了strictAutoFit，没用。我设置maxgrid = 4 如果两个元素，第一个元素设为gridspan=2,第二个设为4，超出了，按理说第二个会被挤下去，但是不会。 主要想实现这个效果

**Reply by beita-wang** [reply]
貌似可以了，感谢
